tests/functional/email_opt_in.js

```diff
@@ -76,90 +63,48 @@ define([
+        .then(openPage(SIGNUP_PAGE_URL, selectors.SIGNUP.HEADER))
         .then(fillOutSignUp(email, PASSWORD, { optInToMarketingEmail: true }))
 
-        .then(testElementExists('#fxa-confirm-header'))
+        .then(testElementExists(selectors.CONFIRM_SIGNUP.HEADER))
         .then(openVerificationLinkInSameTab(email, 0))
 
-        .then(testElementExists('#communication-preferences.basket-ready'))
+        .then(testElementExists(selectors.SETTINGS_COMMUNICATION.READY))
         .then(waitForBasket(email))
-        .then(click('#communication-preferences .settings-unit-toggle'))
-        .then(visibleByQSA('#communication-preferences .settings-unit-details'))
-        .then(testOptedIn())
-
-        // user signed up to basket, so has a preferences URL
-        .then(testElementExists('#preferences-url'))
-
-        // ensure the changes stick across refreshes
-        .refresh()
-
-        .then(testElementExists('#communication-preferences.basket-ready'))
-        .then(visibleByQSA('#communication-preferences .settings-unit-details'))
-        .then(testOptedIn())
-
-        // preference url would open in a new tab
-        .then(testAttributeEquals('#preferences-url', 'target', '_blank'))
+        .then(click(selectors.SETTINGS_COMMUNICATION.MENU_BUTTON))
+        .then(visibleByQSA(selectors.SETTINGS_COMMUNICATION.DETAILS))
 
-        .then(click('#marketing-email-optin'))
-        .then(testSuccessWasShown())
-
-        // ensure the opt-out sticks across refreshes
-        .refresh()
-
-        .then(testElementExists('#communication-preferences.basket-ready'))
-        .then(click('#communication-preferences .settings-unit-toggle'))
-        .then(visibleByQSA('#communication-preferences .settings-unit-details'))
-
-        .then(testNotOptedIn());
+        // user signed up to basket, so has a manage URL
+        .then(testElementExists(selectors.SETTINGS_COMMUNICATION.BUTTON_MANAGE));
     },
 
-    'do not opt-in on signup': function () {
+    'opt-in from settings after signup': function () {
       return this.remote
-        .then(openPage(SIGNUP_PAGE_URL, '#fxa-signup-header'))
+        .then(openPage(SIGNUP_PAGE_URL, selectors.SIGNUP.HEADER))
         .then(fillOutSignUp(email, PASSWORD, { optInToMarketingEmail: false }))
 
-        .then(testElementExists('#fxa-confirm-header'))
+        .then(testElementExists(selectors.CONFIRM_SIGNUP.HEADER))
         .then(openVerificationLinkInSameTab(email, 0))
 
-        .then(testElementExists('#communication-preferences.basket-ready'))
-        .then(click('#communication-preferences .settings-unit-toggle'))
-
-        .then(visibleByQSA('#communication-preferences .settings-unit-details'))
+        .then(testElementExists(selectors.SETTINGS_COMMUNICATION.READY))
+        .then(click(selectors.SETTINGS_COMMUNICATION.MENU_BUTTON))
 
-        .then(testNotOptedIn());
-    },
-
-    'opt in from settings after sign-in': function () {
```

Comment:
This test is more or less equivalent to the sign-up variant, so I collapsed the two tests into one.